lib/history/sqlite.go

```diff
@@ -0,0 +1,379 @@
+/*
+Copyright 2019 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package history
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/jonboulle/clockwork"
+
+	pb "github.com/gravitational/satellite/agent/proto/agentpb"
+
+	"github.com/gravitational/trace"
+	"github.com/jmoiron/sqlx"
+	_ "github.com/mattn/go-sqlite3" // initialize sqlite3
+	log "github.com/sirupsen/logrus"
+)
+
+// SQLiteTimeline represents a timeline of cluster status events. The timeline
+// can hold a specified amount of events and uses a FIFO eviction policy.
+// Timeline events are stored in a local sqlite database.
+//
+// Implements Timeline
+type SQLiteTimeline struct {
+	// clock is used to record event timestamps.
+	clock clockwork.Clock
+	// capacity specifies the max number of events that can be stored in the timeline.
+	capacity int
```

Comment:
Instead of duplicating fields from Config in the struct, we usually just embed the entire Config in the struct:

```go
type Timeline struct {
  Config
  // other fields
}

type Config struct {
  // ...
}
```